examples/grpc-server-with-prometheus/client/client.go

```diff
@@ -1,15 +1,39 @@
 package main
 
 import (
+	"bufio"
 	"fmt"
 	"log"
+	"net/http"
+	"os"
+	"strings"
+	"time"
 
 	"golang.org/x/net/context"
 	"google.golang.org/grpc"
 
 	pb "github.com/grpc-ecosystem/go-grpc-prometheus/examples/grpc-server-with-prometheus/protobuf"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+var (
+	// Create a metrics registry.
+	reg = prometheus.NewRegistry()
+
+	// Create a customized counter metric.
+	customizedCounterMetric = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "demo_client_say_hello_method_call_count",
+		Help: "Total number of RPCs called on the client.",
+	}, []string{"name"})
+)
+
+func init() {
+	// Register customized metrics to registry.
+	reg.MustRegister(customizedCounterMetric)
```

Comment:
this is not a recommended pattern, we should instantiate the registry and register metrics in the main function

Reply:
not using `init` functions or package level/global variables that are mutated (the registry)